src/it/scala/com/databricks/spark/redshift/STSIntegrationSuite.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2015 Databricks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.spark.redshift
+
+import com.amazonaws.auth.BasicAWSCredentials
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient
+import com.amazonaws.services.securitytoken.model.AssumeRoleRequest
+
+import org.apache.spark.sql.{Row, SaveMode}
+import org.apache.spark.sql.types.{IntegerType, StructField, StructType}
+
+/**
+ * Integration tests for accessing S3 using Amazon Security Token Service (STS) credentials.
```

Comment:
Important note here: this class actually _does_ test configuration with a set of STS temporary keys that it requests from a real AWS account (I configured the proper roles and policies to allow for this to be done securely in our dev environment). However, this isn't actually testing the more realistic scenario where the EC2 instance's IAM roles are used to authenticate the Spark <-> S3 connection. As a result, it's possible for this suite's ability to detect bugs to be broken during refactoring, since it's theoretically still possible for the Hadoop configuration's EC2 keys to be passed into Redshift if the test author misspelled the `sts` configuration parameter names.